src/commodore/util.spec.ts

```diff
@@ -32,7 +32,18 @@ describe('src/commodore/util', () => {
       ${'distribution/dist.yml'}             | ${{ distribution: 'dist', cloud: null, region: null }}
       ${'cloud/cloud.yml'}                   | ${{ distribution: null, cloud: 'cloud', region: null }}
       ${'cloud/cloud/region.yml'}            | ${{ distribution: null, cloud: 'cloud', region: 'region' }}
+      ${'cloud/cloud2/region2.yaml'}         | ${{ distribution: null, cloud: 'cloud2', region: 'region2' }}
       ${'cloud/cloud/params.yml'}            | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud2/params.yaml'}          | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud/params/param.yml'}      | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud/foo/bar.yml'}           | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud2/foo/bar/buzz.yaml'}    | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud2/foo/.yaml'}            | ${{ distribution: null, cloud: 'cloud2', region: null }}
+      ${'cloud/cloud//.yml'}                 | ${{ distribution: null, cloud: 'cloud', region: null }}
+      ${'cloud/cloud/.yml'}                  | ${{ distribution: null, cloud: null, region: null }}
+      ${'cloud/cloudnt/.yaml'}               | ${{ distribution: null, cloud: null, region: null }}
```

Comment:
I'm not quite sure what the correct behavior for these cases would be. I mean they are quite esoteric, so I don't know it it matters, but it feels weird to match `cloud/<cloud>/foo/.yml` and not match `cloud/<cloud>/.yaml`.

Reply:
I agree that these cases are not very consistent, but I don't think it matters too much, since users are free to adjust the regex if they need different behaviors. Side-note (not relevant to thit PR): I'd strongly discourage users from actually using `.yml` or `.yaml` as a reclass class (I'm not sure you'd even be able to include such a class...)